aea/connections/base.py

```diff
@@ -19,61 +19,40 @@
 # ------------------------------------------------------------------------------
 
 """The base connection package."""
-
+import asyncio
 from abc import abstractmethod, ABC
-from queue import Queue
-from typing import Optional, TYPE_CHECKING
+from asyncio import AbstractEventLoop
+from typing import TYPE_CHECKING, Optional
 
 from aea.configurations.base import ConnectionConfig
+
 if TYPE_CHECKING:
     from aea.mail.base import Envelope  # pragma: no cover
 
 
-class Channel(ABC):
-    """Abstract definition of a channel."""
-
-    @abstractmethod
-    def connect(self) -> Optional[Queue]:
-        """
-        Set up the connection.
-
-        :return: A queue or None.
-        """
-
-    @abstractmethod
-    def disconnect(self) -> None:
-        """
-        Tear down the connection.
-
-        :return: None.
-        """
-
-    @abstractmethod
-    def send(self, envelope: 'Envelope') -> None:
-        """
-        Send an envelope.
-
-        :param envelope: the envelope to send.
-        :return: None.
-        """
+class AEAConnectionError(Exception):
+    """Exception class for connection errors."""
 
 
 class Connection(ABC):
     """Abstract definition of a connection."""
 
-    channel: Channel
-
-    def __init__(self):
+    def __init__(self, connection_id: str, loop: Optional[AbstractEventLoop] = None):
```

Comment:
Should we not make the `loop` required?

Reply:
To provide a simple instantiation we fetch the default event loop if none is provided.